src/openvic-simulation/modifier/ModifierEffectCache.cpp

```diff
@@ -196,7 +200,7 @@ ModifierEffectCache::ModifierEffectCache()
 	/* Military Modifier Effects */
 	attack { nullptr },
```

Comment:
attack_leader for consistency

Reply:
IMO it's more consistent to use `attack` - these variable names correspond to the `ModiferEffect`s' identifier strings, so `attack` has identifier `"attack"` and `defence_leader` has identifier `"defence leader"`. This makes it clear exactly what the variable corresponds to, including the fact that the `attack` effect could technically appear in places other than leader traits while `defence_leader` is guarunteed to only come from leader traits (we may want to enforce similar restrictions for effects like `attack`, but that can come in a later PR which can change the variable names too if necessary).

Reply:
My comment wasn't about the origin of the modifier, it is about the effect of the modifier.
attack means a boost to the roll from the leader. This is how it shows up in the UI. It's unrelated to attack stats for units.
Thus attack technically can't appear in places other than leader background/traits as it is scoped to the leader.

Reply:
I mentioned the modifier's origin because that's what the modifier effect cache variable names are currently based on, so they provide useful information about what the variable actually means (IMO the origin information is more useful because that's dependent on the code and is difficult to manually determine, whereas the use information is fixed regardless of the game's code/implementation).
And while it's correct that the `attack` modifier effect shouldn't appear anywhere other than leader traits, the current implementation doesn't force that, meaning `attack` effects in country and province modifiers will be parsed and included in modifier sums. Whether these potential misuses of the effect actually get used in battle logic depends on its implementation, but we should probably restrict the effect's use regardless. When we have such a restriction we could change the variable's name to represent the fact that it definitely only appears in leader traits, but until then the current name serves as a warning that the variable just represents the `attack` effect without any context restrictions.

Reply:
I think there could be arguments for allowing attack modifiers from other origins:

- +1 attack when fighting on core (similar to hoi4)
- military academy building
- school of the attack invention
- military industrial complex tech school

Just some ideas that might be interesting for a mod.
The point is that attack in all these cases means attack bonus from leader in the UI, not gas attack or some other modifier.